test/unittest/unittest_nntrainer_layers.cpp

```diff
@@ -601,6 +602,79 @@ TEST(nntrainer_Pooling2D, forwarding_01_p) {
   }
 }
 
+/**
+ * @brief Flatten Layer
+ */
+TEST(nntrainer_Flatten, forwarding_01_p) {
+  int status = ML_ERROR_NONE;
+  nntrainer::FlattenLayer layer;
+  std::vector<std::string> input_str;
+  nntrainer::TensorDim previous_dim;
+  previous_dim.setTensorDim("1:2:4:4");
+  layer.setInputDimension(previous_dim);
+
+  status = layer.initialize(false);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  nntrainer::Tensor in(1, 2, 4, 4);
+  nntrainer::Tensor out, result(1, 1, 1, 32);
+  std::ifstream file("goldenPooling2DResult.out");
+  in.read(file);
+  file.close();
+  out = layer.forwarding(in, status);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  EXPECT_EQ(out.getDim().batch(), 1);
+  EXPECT_EQ(out.getDim().channel(), 1);
+  EXPECT_EQ(out.getDim().height(), 1);
+  EXPECT_EQ(out.getDim().width(), 32);
+
+  float *out_ptr, *golden;
+  golden = in.getData();
+  out_ptr = out.getData();
+
+  for (int i = 0; i < 32; ++i) {
+    EXPECT_FLOAT_EQ(out_ptr[i], golden[i]);
+  }
+}
+
+/**
+ * @brief Flatten Layer
+ */
+TEST(nntrainer_Flatten, backwarding_01_p) {
+  int status = ML_ERROR_NONE;
+  nntrainer::FlattenLayer layer;
+  std::vector<std::string> input_str;
+  nntrainer::TensorDim previous_dim;
+  previous_dim.setTensorDim("1:2:4:4");
+  layer.setInputDimension(previous_dim);
+
+  status = layer.initialize(false);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  nntrainer::Tensor in(1, 1, 1, 32);
+  nntrainer::Tensor out, result(1, 1, 1, 32);
```

Comment:
how about also creating a unit test with batch size more than 1 to test that flatten does not affect the batch size?

Reply:
You are right it does not affect the batch. However the more unit test is the better. So I will add more test cases. ( and negative test cases as well )

Reply:
some more unit tests are added except negative test cases.